### PR TITLE
Clean up OpenCL compilation into SPIR-V.

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
This change removes any references to Arm in SPIR-V compiler for OpenCL as there isn't anything specific to Arm in the compilation process. It also avoids using Arm specific workaround because SPIR-V compilation in clang doesn't suffer from the same limitation.

This change also makes explicit the fact that compilation uses llvm-spirv as clang now has internal SPIR-V support too which should be added here once it matures.

### CODE REVIEW

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Hi! Thanks for this.
Unfortunately as this changes compiler ids, we can't merge as-is, as old links with these compilers would break.
To fix this, either leave the old ids (Which maybe not the best idea given that removing arm references is the point of this PR), or for every changed compiler id, add a prop like:
`compiler.<newid>.alias=<oldid>` so that old links will be able to still find the compiler

**Comment by AnastasiaStulova** [comment]
> Hi! Thanks for this. Unfortunately as this changes compiler ids, we can't merge as-is, as old links with these compilers would break. To fix this, either leave the old ids (Which maybe not the best idea given that removing arm references is the point of this PR), or for every changed compiler id, add a prop like: `compiler.<newid>.alias=<oldid>` so that old links will be able to still find the compiler

Thanks for the suggestion. I have added alias to all the compilers being renamed.

File: etc/config/cpp_for_opencl.amazon.properties
Comment (RubenRBS) [inline etc/config/cpp_for_opencl.amazon.properties:185]: As far as I can see, your PR fixing this hasn't yet been merged, right? (LLVM's issue tracker is a bit hard to read, so sorry if I missed something) Do we not need the workaround anymore?

**Review by RubenRBS — APPROVED** [review]
Changes look great now, thanks so much for taking the time to fix the id issue. I have one final question regarding the default options before we merge

File: etc/config/cpp_for_opencl.amazon.properties
Comment (mattgodbolt) [reply etc/config/cpp_for_opencl.amazon.properties:185]: @AnastasiaStulova thanks again for your contribution: we'd love to merge. Can you answer @RubenRBS 's question so we know where we are on this?

Comment (AnastasiaStulova) [reply etc/config/cpp_for_opencl.amazon.properties:185]: Sorry for delay, this clang bug is only for compiling to Arm targets. When we compile for SPIR/SPIR-W: `-triple spir`, `-triple spir64`, `-triple spirv32` or `-triple spirv64` no workaround is needed as clang just compiles correctly.

**Comment by mattgodbolt** [comment]
Thanks: I'm going to merge this so we don't get too stale. Sorry for the delay.

**Pull request merged**
